### PR TITLE
Time values should be unspecified when converting Date using ToDateTime

### PR DESCRIPTION
**Pull request by dczulada**
https://cql.hl7.org/04-logicalspecification.html#todatetime
For Date values, the result is a DateTime with the time components unspecified, except the timezone offset, which is set to the timezone offset of the evaluation request timestamp.

Previously the time components are being set to 0.

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository.
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [x] Code passes lint and prettier (hint: use `yarn run test:plus` to run tests, lint, and prettier)
- [x] All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [x] `cql4browsers.js` built with `yarn run build:browserify` if source changed.

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by p9g** [comment]
Related to errata in CQL spec https://jira.hl7.org/browse/FHIR-35869

File: src/datatypes/datetime.ts [hunk 1]
```diff
@@ -1179,7 +1179,7 @@ export class Date extends AbstractDate {
     // except for the timezone offset, which will be set to the timezone offset of the evaluation
     // request timestamp. (this last part is acheived by just not passing in timezone offset)
     if (this.year != null && this.month != null && this.day != null) {
-      return new DateTime(this.year, this.month, this.day, 0, 0, 0, 0);
+      return new DateTime(this.year, this.month, this.day, null, null, null, null);
```
Comment (cmoesel) [inline src/datatypes/datetime.ts:1182]: Line 1178 above (which GitHub will not let me directly comment on) says:
```ts
// from the spec: the result will be a DateTime with the time components set to zero,
```

That comment is now out of sync w/ the implementation.  Can you update it to reflect the correct behavior? It should read like this now:
```ts
// from the spec: the result will be a DateTime with the time components unspecified,
```

Also, while you're at it, could you fix the typo on line 1180 (`acheived`)?

File: test/elm/comparison/comparison-test.ts [hunk 2]
```diff
@@ -71,20 +71,20 @@ describe('Equal', () => {
     this.impossiblyEqualDateTimes.exec(this.ctx).should.be.false();
   });
 
-  it('should be null for Date and DateTime equality with same year, month, hour', function () {
-    should(this.dateAndDateTimeNull.exec(this.ctx)).be.null();
+  it('should be true for Date and DateTime equality with same year, month, hour', function () {
+    this.dateAndDateTimeNull.exec(this.ctx).should.be.true();
```
Comment (cmoesel) [inline test/elm/comparison/comparison-test.ts:75]: The name of the CQL expression (and variable here) is now out of sync w/ the expectation.  Now that this returns `true`, the CQL expression should be `DateAndDateTimeTrue` (and corresponding variable here should be `dateAndDateTimeTrue`).

All of the test expressions and variables need to be checked and fixed to ensure their name still matches the expectation. (So I won't comment on anymore like this).

**Review by cmoesel — CHANGES_REQUESTED** [review]
Thanks for this fix, @dczulada!  Everything seems to work great (and as expected).  I just found a few places where a comment and/or test expression or variable name needs to be changed.

**Review by cmoesel — APPROVED** [review]
Looks good to me!  Thanks, @dczulada!

**Review by mgramigna — APPROVED** [review]
Looks good and passed all my tests!

**Pull request merged**